### PR TITLE
Moientajik/EventIds refactor

### PR DESCRIPTION
**Pull request by MoienTajik**
- Used **nameof** in place of string ([CA1507](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1507?view=vs-2019)) in _EventIds.cs_
- Fixed typo in _EventIds_: "Deteced" -> "Detected"
- Renamed some event-id name parameters to match the field name

### CODE REVIEW

File: src/ReverseProxy/EventIds.cs [hunk 1]
```diff
@@ -7,46 +7,46 @@ namespace Microsoft.ReverseProxy
...
-        public static readonly EventId AffinityCannotBeEstablishedBecauseNoDestinationsFoundOnCluster = new EventId(36, "AffinityCannotBeEstablishedBecauseNoDestinationsFoundOnCluster");
-        public static readonly EventId NoDestinationOnClusterToEstablishRequestAffinity = new EventId(37, "NoDestinationOnClusterToEstablishRequestAffinity");
-        public static readonly EventId RequestAffinityKeyDecryptionFailed = new EventId(38, "RequestAffinityKeyDecryptionFailed");
-        public static readonly EventId DestinationMatchingToAffinityKeyNotFound = new EventId(39, "DestinationMatchingToAffinityKeyNotFound");
-        public static readonly EventId RequestAffinityHeaderHasMultipleValues = new EventId(40, "RequestAffinityHeaderHasMultipleValues");
-        public static readonly EventId AffinityResolutionFailureWasHandledProcessingWillBeContinued = new EventId(41, "AffinityResolutionFailureWasHandledProcessingWillBeContinued");
+        public static readonly EventId ApplyProxyConfig = new EventId(1, nameof(ApplyProxyConfig));
```
Comment (Kahbazi) [inline src/ReverseProxy/EventIds.cs:10]: > I would avoid nameof here. 

https://github.com/dotnet/aspnetcore/pull/11379#discussion_r295441810

Comment (MoienTajik) [reply src/ReverseProxy/EventIds.cs:10]: hmm, are those mentioned reasons apply here too?

Comment (Kahbazi) [reply src/ReverseProxy/EventIds.cs:10]: In my opinion yes, both refer to name for `EventId`.

Comment (Tratcher) [reply src/ReverseProxy/EventIds.cs:10]: Correct, please revert this change.

Comment (MoienTajik) [reply src/ReverseProxy/EventIds.cs]: Ok, changes reverted and I opened a new pull-request: https://github.com/microsoft/reverse-proxy/pull/320

**Pull request closed without merging**